src/main/java/ch/njol/skript/expressions/ExprMessage.java

```diff
@@ -1,225 +1,279 @@
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.EnumSet;
+
+import org.bukkit.event.Event;
+import org.bukkit.event.command.UnknownCommandEvent;
+import org.bukkit.event.entity.EntityDeathEvent;
+import org.bukkit.event.entity.PlayerDeathEvent;
+import org.bukkit.event.player.AsyncPlayerChatEvent;
+import org.bukkit.event.player.PlayerChatEvent;
+import org.bukkit.event.player.PlayerJoinEvent;
+import org.bukkit.event.player.PlayerKickEvent;
+import org.bukkit.event.player.PlayerQuitEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.ScriptLoader;
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Events;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.events.util.PlayerChatEventHandler;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.log.ErrorQuality;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+/**
+ * @author Peter Güttinger
+ */
+@SuppressWarnings("deprecation")
+@Name("Message")
+@Description("The (chat) message of a chat event, the join message of a join event, the quit message of a quit event, or the death message on a death event. This expression is mostly useful for being changed.")
+@Examples({"on chat:",
+	"\tplayer has permission \"admin\"",
+	"\tset message to \"<red>%message%\"",
+	"",
+	"on first join:",
+	"\tset join message to \"Welcome %player% to our awesome server!\"",
+	"",
+	"on join:",
+	"\tplayer has played before",
+	"\tset join message to \"Welcome back, %player%!\"",
+	"",
+	"on quit:",
+	"\tset quit message to \"%player% left this awesome server!\"",
+	"",
+	"on death:",
+	"\tset the death message to \"%player% died!\"",
+	"",
+	"on unknown command: #requires Paper",
+	"\tset unknown command message to \"The command \"\"%command%\"\" doesn't exist.\""})
+@Since("1.4.6 (chat message), 1.4.9 (join & quit messages), 2.0 (death message), INSERT VERSION (unknown command message)")
+@Events({"chat", "join", "quit", "death", "unknown command"})
+public class ExprMessage extends SimpleExpression<String> {
+	
+	static {
+		Skript.registerExpression(ExprMessage.class, String.class, ExpressionType.SIMPLE, MessageType.patterns.toArray(new String[]{}));
+	}
+	
+	@SuppressWarnings("null")
+	private MessageType type;
+	
+	
+	@SuppressWarnings("null")
+	@Override
+	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
+		type = (MessageType) MessageType.supportedTypes.toArray()[matchedPattern];
```

Comment:
`toArray` not needed here - you can just use `get(matchedPattern)`

Reply:
I already tried but get apparently isn't an EnumSet method :-/

Reply:
this is because sets are not guaranteed order